src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,225 @@
+                    case "delete":
+                        try {
+                            deleteEvent(userLists, splitInputs, numOfTasks);
+                            numOfTasks--;
+                        } catch (ArrayIndexOutOfBoundsException | NullPointerException e){
+                            System.out.println(TEXT_SPACER + "Please choose a valid entry to delete.");
+                        }
+                        break;
+                    default:
+                        throw new DukeExceptions(TEXT_SPACER +
+                                "Sorry mate, I don't understand what you are trying to get me to do.");
+                    }
+
+                    saveToFile(userLists, numOfTasks);
+                } catch (DukeExceptions | IOException e) {
+                    System.out.println(e.getMessage());
+                }
+                printLineSpacer();
+
+            }
+        } while(!userLineInput.equals("bye"));
+
+        System.out.println(TEXT_SPACER + "Aight. See you soon mate.");
+        printLineSpacer();
+    }
+
+    private static int getFile(ArrayList userLists, int numOfTasks) {
+        try {
+            File saveFile = new File(path + "/task.txt");
+            saveFile.createNewFile();
+            Scanner savedTasks = new Scanner(saveFile);
+            while(savedTasks.hasNext()) {
+                String[] currentLine = savedTasks.nextLine().split("\\|");
+                String taskType = currentLine[0];
+                String taskStatus = currentLine[1];
+                String taskCommand;
+                switch (taskType) {
+                case "T ":
+                    taskCommand = "todo" + currentLine[2];
+                    storeToDo(userLists, taskCommand, numOfTasks);
+                    break;
+                case "D ":
+                    taskCommand = "deadline" + currentLine[2];
+                    storeDeadline(userLists, taskCommand, numOfTasks);
+                    break;
+                case "E ":
+                    taskCommand = "event" + currentLine[2];
+                    storeEvent(userLists, taskCommand , numOfTasks);
+                    break;
+                }
+                if(taskStatus.equals("1")){
+                    ((Task)userLists.get(numOfTasks)).markAsDone();
+                }
+                numOfTasks++;
+                }
+
+        } catch (IOException e) {
+            System.out.println(TEXT_SPACER + "No save file. New file will be created");
+        } catch (DukeExceptions e){
+            System.out.println(e.getMessage());
+        }
+        return numOfTasks;
+    }
+
+    public static void saveToFile(ArrayList userList, int numTask) throws IOException {
+        FileWriter updateFile = new FileWriter(path + "/task.txt");
+        //TO FIX: DOESN'T UPDATE DONE STATUS
+        for(int i = 0; i < numTask; i++) {
+            String stringToSave = ((Task)userList.get(i)).getTaskType() + " | " + getDoneStatus(userList, i) +
```

Comment:
This string looks complex. How about forming it step by step using variable declaration and [Java String format()](https://www.javatpoint.com/java-string-format)?